control/grpc.py

```diff
@@ -17,30 +17,45 @@
 import threading
 import errno
 import contextlib
+from typing import Callable
+from collections import defaultdict
 
 import spdk.rpc.bdev as rpc_bdev
 import spdk.rpc.nvmf as rpc_nvmf
 import spdk.rpc.log as rpc_log
 from spdk.rpc.client import JSONRPCException
 
 from google.protobuf import json_format
+from google.protobuf.empty_pb2 import Empty
 from .proto import gateway_pb2 as pb2
 from .proto import gateway_pb2_grpc as pb2_grpc
+from .proto import monitor_pb2
+from .proto import monitor_pb2_grpc
 from .config import GatewayConfig
 from .utils import GatewayEnumUtils
 from .utils import GatewayUtils
 from .utils import GatewayLogger
-from .state import GatewayState
+from .state import GatewayState, GatewayStateHandler, OmapLock
 from .cephutils import CephUtils
 
-MAX_ANA_GROUPS = 4
+MAX_ANA_GROUPS = 16
```

Comment:
I think we need max 32 ana groups, because we can have up to 32 GWs in a GW group.

Reply:
Currently the value is matching the [c++ definition](https://github.com/ceph/ceph-ci/blob/8306a471398146176678a5538a4cf59582c0948d/src/mon/NVMeofGwTypes.h#L41)
```c++
#define MAX_SUPPORTED_ANA_GROUPS 16
```
this constant could be removed, MAX NAMESPACES for the subsystem could be used instead.

Reply:
✅ removed the constant